plugin/src/main/java/spp/jetbrains/sourcemarker/VariableParser.java

```diff
@@ -0,0 +1,81 @@
+package spp.jetbrains.sourcemarker;
+
+import com.intellij.openapi.util.Pair;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static spp.jetbrains.sourcemarker.Constants.EMPTY;
+import static spp.jetbrains.sourcemarker.Constants.PATTERN_CURLY_BRACES;
+
+public class VariableParser {
```

Comment:
This class is not thread-safe. It should be stateless

Reply:
The class is acting as a container for existing variables and its related methods and the object is not shared. Looks fine to me. Please suggest.

Reply:
A bit odd you're not aware of thread-safety. See: https://www.baeldung.com/java-thread-safety

Reply:
Resolved. Please check